common/components/SettingsForm.tsx

```diff
@@ -2635,14 +2647,36 @@ export default function SettingsForm({
                                     onChange={(event) => handleSettingChanged('tabName', event.target.value)}
                                 />
                             )}
+                            <TextField
+                                select
+                                label={t('settings.regexPreset')}
+                                value={selectedRegexPresetName}
+                                color="primary"
+                                onChange={(event) => {
+                                    let presetSelected = regexPresetsList.find((x) => x.name === event.target.value);
+                                    setSelectedRegexPresetName(presetSelected!.name);
+                                    handleSettingChanged('subtitleRegexFilter', presetSelected!.regex);
+                                }}
+                            >
+                                {regexPresetsList.map((s) => (
+                                    <MenuItem key={s.name} value={s.name}>
+                                        {s.name}
+                                    </MenuItem>
+                                ))}
+                            </TextField>
                             <TextField
                                 label={t('settings.subtitleRegexFilter')}
                                 fullWidth
                                 value={subtitleRegexFilter}
                                 color="primary"
                                 error={!validRegex}
                                 helperText={validRegex ? undefined : 'Invalid regular expression'}
-                                onChange={(event) => handleSettingChanged('subtitleRegexFilter', event.target.value)}
+                                onChange={(event) => {
+                                    handleSettingChanged('subtitleRegexFilter', event.target.value);
+                                    if (selectedRegexPresetName !== regexPresetsList[0].name) {
+                                        setSelectedRegexPresetName(regexPresetsList[0].name);
```

Comment:
It's risky to update state from inside the render function. You should consider using `useEffect` instead.
Also, ideally if the user types in one of the presets manually the correct one gets selected. This code is selecting the first preset no matter what.